### PR TITLE
19050 Added pending dissolution check

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#19050

*Description of changes:*
- app version = 5.6.35
- added check for pending dissolution filing
- moved checks as new functions in mixin
- added business status (message) for pending dissolution
- added enum value
- updated state interface
- added new rule in mixin
- misc clean and missing code
- updated unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Amalgamation/BusinessStatus.vue [hunk 1]
```diff
@@ -90,6 +90,9 @@ export default class BusinessStatus extends Vue {
         return 'This business is not in good standing. This filing cannot be submitted until all ' +
           'businesses are in good standing.'
 
+      case AmlStatuses.ERROR_PENDING_DISSOLUTION_FILING:
+        return 'This business has a pending dissolution filing. It cannot be part of an amalgamation.'
```
Comment (severinbeauvais) [inline src/components/Amalgamation/BusinessStatus.vue:94]: Tooltip is approve by Yui.

File: src/mixins/amalgamation-mixin.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export default class AmalgamationMixin extends Vue {
     this.notInGoodStanding,
     this.limitedRestoration,
     this.futureEffectiveFiling,
+    this.pendingDissolutionFiling,
```
Comment (severinbeauvais) [inline src/mixins/amalgamation-mixin.ts:31]: Note that we check future-effective before pending-resolution, per Yui:

image

File: src/mixins/amalgamation-mixin.ts [hunk 3]
```diff
@@ -166,20 +175,16 @@ export default class AmalgamationMixin extends Vue {
...
+        LegalServices.fetchFirstOrOnlyFiling(item.identifier)
+      ]).then(results => results.map((result: any) => result.value || null))
 
-    return {
-      authInfo: data[0],
-      businessInfo: data[1],
-      addresses: data[2],
-      firstFiling: data[3]
-    }
+    return { authInfo, businessInfo, addresses, firstFiling }
```
Comment (severinbeauvais) [inline src/mixins/amalgamation-mixin.ts:187]: Fancy code, cause it looks nicer 😁

File: src/mixins/amalgamation-mixin.ts [hunk 4]
```diff
@@ -200,9 +205,35 @@ export default class AmalgamationMixin extends Vue {
...
+  /**
+   * This business is future effective if the first filing in the ledger is future effective and is still
+   * not complete or corrected (ie, it's paid or pending).
+   * @param business The business to check if is Future Effective or not.
+   */
+  isFutureEffective (business: any): boolean {
+    return (
+      business.firstFiling?.isFutureEffective === true &&
+      business.firstFiling?.status !== FilingStatus.COMPLETED &&
+      business.firstFiling?.status !== FilingStatus.CORRECTED
```
Comment (severinbeauvais) [inline src/mixins/amalgamation-mixin.ts:217]: Note the small change from before -- previous we check status for PAID or PENDING. This new check is consistent with what we do in [Filings UI](https://github.com/bcgov/business-filings-ui/blob/4f6698f3f4fb6e95fef6c4695ac3abb31b2afb21/src/components/Dashboard/FilingHistoryList/HeaderActions.vue#L123) (ie, reuse business logic we already thought about).

File: src/mixins/amalgamation-mixin.ts [hunk 5]
```diff
@@ -222,11 +253,12 @@ export default class AmalgamationMixin extends Vue {
           role: AmlRoles.AMALGAMATING,
           identifier: tingBusiness.businessInfo.identifier,
           name: tingBusiness.businessInfo.legalName,
-          email: tingBusiness.authInfo?.contacts[0].email || null,
+          email: tingBusiness.authInfo?.contacts[0].email,
           legalType: tingBusiness.businessInfo.legalType,
-          address: tingBusiness.addresses?.registeredOffice.mailingAddress || null,
```
Comment (severinbeauvais) [inline src/mixins/amalgamation-mixin.ts:227]: I don't remember why we added these fallbacks. I think we don't need them -- the property will be "falsy" either way.

On the other hand, if we do need the fallbacks, then they should be added to the [Amalgamating Businesses component](https://github.com/bcgov/business-create-ui/blob/50822182b2c6936ff5341f0aa488a99256f9cd2a/src/components/Amalgamation/AmalgamatingBusinesses.vue#L376) as well. LMK

File: src/mixins/amalgamation-mixin.ts [hunk 6]
```diff
@@ -222,11 +253,12 @@ export default class AmalgamationMixin extends Vue {
...
+          address: tingBusiness.addresses?.registeredOffice.mailingAddress,
           isNotInGoodStanding: (tingBusiness.businessInfo.goodStanding === false),
-          isFutureEffective: (tingBusiness.firstFiling.isFutureEffective === true),
+          isFutureEffective: this.isFutureEffective(tingBusiness),
+          isPendingDissolution: this.isPendingDissolution(tingBusiness),
```
Comment (severinbeauvais) [inline src/mixins/amalgamation-mixin.ts:261]: Now calling the same functions as from AmalgamatingBusinesses.vue.

File: src/mixins/amalgamation-mixin.ts (hunk 5 above)
Comment (JazzarKarim) [reply src/mixins/amalgamation-mixin.ts:227]: I think it was here Sev: https://github.com/bcgov/business-create-ui/pull/608#discussion_r1442107071

I'm OK with this.

**Review by JazzarKarim — APPROVED** [review]
LGTM!

File: src/mixins/amalgamation-mixin.ts (hunk 5 above)
Comment (severinbeauvais) [reply src/mixins/amalgamation-mixin.ts:227]: Thanks for finding that old review comment. I still think it's not needed. Resolving.

**Pull request merged**